.github/workflows/dist.yml

```diff
@@ -0,0 +1,71 @@
+# when `latest` tag or another tag starting with `v` is pushed,
+# this workflow runs to push the distrubtion into a `dist-<tag>`
+# tag
+name: DISTRIBUTE
+
+on:
+  push:
+    tags:
+      - latest
+      - v*
+
+jobs:
+  distribute:
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: "checkout"
+        uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      # always test before dist
+      - uses: eskatos/gradle-command-action@v1
+        name: Build and Test
+        with: 
+          build-root-directory: AndroidXCI
+          wrapper-directory: AndroidXCI
+          wrapper-cache-enabled: true
+          dependencies-cache-enabled: true
+          configuration-cache-enabled: true
+          arguments: check :ftlModelBuilder:check
+      # build distribution
+      - uses: eskatos/gradle-command-action@v1
+        name: Build Distribution
+        with: 
+          build-root-directory: AndroidXCI
+          wrapper-directory: AndroidXCI
+          wrapper-cache-enabled: true
+          dependencies-cache-enabled: true
+          configuration-cache-enabled: true
+          arguments: :cli:installDist
+      # push it to a new git tag
+      - name: "create dist branch"
+        id: event-args
+        run: |
+          set -x
+          REF="${{ github.event.ref }}"
+          TAG=$(echo $REF| sed 's/refs\/tags\///')
+          DIST_TAG="dist-$TAG"
+          git config user.email "${{ github.event.head_commit.author.email }}"
+          git config user.name "${{ github.event.head_commit.author.name }}"
+          git checkout -b tmp-$DIST_TAG
+          # remove workflow etc, otherwise we'll need workflow override permissions
+          rm -rf .github
+          git add .github
+
+          # replace src build command with dist
+          sed '/\#SRC/!b;n;d' action.yml| sed 's/\#DIST//'| sed '/SRC/d' > action.yml
```

Comment:
To make this a bit more readable you could try:

actions.yml
```
    # dist.yml script swaps the following when building distribution
    #REPLACE_BELOW_WITH- run: ${{ github.action_path }}/dist/bin/cli
    - run: cd ${{ github.action_path }}/AndroidXCI && ./gradlew :cli:run
```

dist.yml
```
REPLACE_TOKEN="#REPLACE_BELOW_WITH"
sed -i "/$REPLACE_TOKEN/{n;d}" action.yml
sed -i "s/$REPLACE_TOKEN//g" action.yml
```

Reply:
that looks better, updated (used `REPLACE_NEXT_LINE`)